### ISSUE TITLE
Make View's background_color attribute a property like Window's

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#2469]
### What should be added/changed?
**TL;DR:** Make `View` handle background colors with validation like `Window`

In `Window`, we have a getter and setter for [`Window.background_color`](https://github.com/pythonarcade/arcade/blob/33c368d50c34f67565001a933e8b8bbe9609e656/arcade/application.py#L396-L398):
https://github.com/pythonarcade/arcade/blob/33c368d50c34f67565001a933e8b8bbe9609e656/arcade/application.py#L396-L398

It seems worthwhile to:
1. Add the same behavior to `View`
2. Add tests to ensure homogeneous behavior

### What would it help with?
1. API consistency
2. If @einarf approves making the default color for both `BLACK` instead of `TRANSPARENT_BLACK`:
   1. Screenshots will no longer be surprise-transparent
   2. Making screenshots for examples will be nicer

**Comment by einarf** [comment]
It makes sense that the `View` can have its own background color. This background color should only be used when the user calls `View.clear()` without any arguments.

**Comment by DragonMoffon** [comment]
So, until we get a functioning pixel testing feature on the Window, we can't test this super concretely, but the rest of it has been implemented in #2470

**Comment by eruvanos** [comment]
Merged

**Comment by DragonMoffon** [comment]
Competed with #2470
